src/data/formfactors.h

```diff
@@ -24,9 +24,10 @@
 #include "base/enumoptions.h"
 #include "data/elements.h"
 #include "data/formfactordata.h"
+#include <optional>
 #include <tuple>
 
-template <class T> using optional = std::tuple<T, bool>;
+template <class T> using opt = std::optional<std::reference_wrapper<T>>;
```

Comment:
```suggestion
template <class T> using optional_reference_wrapper = std::optional<std::reference_wrapper<T>>;
```

`optional_reference_wrapper` is clearer than just `opt` but still shorter.